File: GridSegmantation/GridSegmentation/Grid.cs
using System;

namespace GridSegmentation
{
  public class Grid
  {
    public int iCount { get; private set; }
    public int jCount { get; private set; }
    public int cellsCount { get; private set; }
    public int[] cells { get; private set; }

    public Grid(int iCount = 0, int jCount = 0)
    {
      this.Init(iCount, jCount);
    }

    public void Init(int iCount, int jCount)
    {
      this.iCount = iCount;
      this.jCount = jCount;
      this.cellsCount = this.iCount * this.jCount;

      this.cells = new int[this.cellsCount];
    }

    public int Index(int i, int j)
    {
      return j * this.iCount + i;
    }
    public void Index(int index, out int i, out int j)
    {
      i = index % this.iCount;
      j = index / this.iCount;
    }

    public void InitCells(int filledCountPercents, int value = 1)
    {
      Array.Clear(this.cells, 0, this.cellsCount);

      int filledCount = filledCountPercents * this.cellsCount / 100;
      for (int i = 0; i < filledCount; ++i)
      {
        this.cells[this.Index(Utils.Random.Next(this.iCount), Utils.Random.Next(this.jCount))] = value;
      }
    }
  }
}

File: capton/grid.py
class Grid:
	def __init__(self, i_count = 0, j_count = 0):
		self.i_count = i_count
		self.j_count = j_count
		self.cells = [[None for j in range(self.j_count)] for i in range(self.i_count)]

	def clear(self):
		for i in range(self.grid.i_count):
			for j in range(self.grid.j_count):
				self.cells[i][j] = None

	def check_range(self, i, j):
		return 0 <= i < self.i_count and 0 <= j < self.j_count

	def is_empty(self):
		return self.i_count == 0 and self.j_count == 0


def find_none_random(grid):
	while True:
		i = random.randint(0, grid.i_count - 1)
		j = random.randint(0, grid.j_count - 1)
		if grid.cells[i][j] is None:
			return (i, j)

def fill_random(grid, fill_percent):
	grid.clear()

	fill_count = (int)(self.grid.i_count * self.grid.j_count * fill_percent / 100)
	fill_index = 0
	while fill_index < fill_count:
		(i, j) = find_none_random(grid)
		grid.cells[i][j] = True
		fill_index += 1

